### PR TITLE
Fixed Shield Dust, added tests

### LINKED ISSUE #4104: Shield Dust doesn't work
### Description

Title. In my test case, I used Trop Kick on a Ribombee with Shield Dust and it dropped it's attack. The Ability might altogether be broken.

### Version

upcoming (Edge)

### Upcoming/master Version

09d12fb

### Discord contact info

_No response_

### PR DESCRIPTION
**Pull request by cfmnephrite**
Also fixed a duplicate macro caused by near-simultaneous PR merges (oops)

## Issue(s) that this PR fixes
#4104 

## Feature(s) this PR does NOT handle:
Does not implement Sparkling Aria's peculiar mechanic whereby its effect is NOT blocked by Shield Dust when it hits multiple mons in doubles.

## **Discord contact info**
The Cancer Fairy/thechurchofcage

### CODE REVIEW

File: test/battle/ability/shield_dust.c [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+                ANIMATION(ANIM_TYPE_GENERAL, B_ANIM_STATS_CHANGE, player);
+                break;
+            case MOVE_METEOR_ASSAULT: // second turn
+                MESSAGE("Wobbuffet must recharge!");
+                break;
+        }
+    }
+}
+
+TO_DO_BATTLE_TEST("Shield Dust does not block Sparkling Aria in doubles"); // Would certainly fail at the moment
```
Comment (DizzyEggg) [inline test/battle/ability/shield_dust.c:124]: Let's create a known failing for this

**Review by DizzyEggg — CHANGES_REQUESTED** [review]

File: test/battle/ability/shield_dust.c (hunk 1 above)
Comment (cfmnephrite) [reply test/battle/ability/shield_dust.c:124]: Oof, I don't really want to as part of this, tbh - this is just a bugfix PR for #4104 . We can open an issue for it, but it's such a minor mechanic...

Comment (AlexOn1ine) [reply test/battle/ability/shield_dust.c:124]: Let me write one fast so we don't waste an issue

Comment (cfmnephrite) [reply test/battle/ability/shield_dust.c:124]: > Let me write one fast so we don't waste an issue

Appreciate it :)

Comment (AlexOn1ine) [reply test/battle/ability/shield_dust.c:124]: Isn't this an easy fix, or is there a problem with excluding Sparkling Aria effect?
```
DOUBLE_BATTLE_TEST("Shield Dust does not block Sparkling Aria in doubles")
{
    KNOWN_FAILING;
    GIVEN {
        PLAYER(SPECIES_WOBBUFFET);
        PLAYER(SPECIES_WOBBUFFET);
        OPPONENT(SPECIES_VIVILLON) { Ability(ABILITY_SHIELD_DUST); Status1(STATUS1_BURN); }
        OPPONENT(SPECIES_WOBBUFFET);
    } WHEN {
        TURN { MOVE(playerLeft, MOVE_SPARKLING_ARIA); }
    } SCENE {
        ANIMATION(ANIM_TYPE_MOVE, MOVE_SPARKLING_ARIA, playerLeft);
        NOT MESSAGE("Foe Vivillion's burn was healed.");
        STATUS_ICON(opponentLeft, burn: TRUE);
    }
}
```

Comment (AlexOn1ine) [reply test/battle/ability/shield_dust.c:124]: Then add the test to the PR so it can be merged 😅

**Comment by AlexOn1ine** [comment]
Egg, CF added the test I provided. Can this be merged? :eyes:

**Comment by DizzyEggg** [comment]
> Egg, CF added the test I provided. Can this be merged? 👀

Yes, go ahead

**Review by DizzyEggg — APPROVED** [review]

**Pull request merged**